diff -ruN media_entity_audio_patch_3/media_entity_audio.module media_entity_audio_patch_4/media_entity_audio.module
--- media_entity_audio_patch_3/media_entity_audio.module	2017-11-06 14:17:16.824073771 +1100
+++ media_entity_audio_patch_4/media_entity_audio.module	2017-11-06 14:18:04.659263612 +1100
@@ -12,7 +12,7 @@
  */
 function media_entity_audio_theme() {
   return array(
-    'media_file_formatter' => array(
+    'media_audio_file_formatter' => array(
       'variables' => array(
         'file' => NULL,
         'description' => NULL,
@@ -25,9 +25,9 @@
 }
 
 /**
- * Implements template_preprocess_media_file_formatter().
+ * Implements template_preprocess_media_audio_file_formatter().
  */
-function media_entity_audio_preprocess_media_file_formatter(&$vars) {
+function media_entity_audio_preprocess_media_audio_file_formatter(&$vars) {
   $file = $vars['file'];
   $file_entity = ($file instanceof File) ? $file : File::load($file->fid);
   $url = file_create_url($file_entity->getFileUri());
diff -ruN media_entity_audio_patch_3/src/Plugin/Field/FieldFormatter/AudioPlayerHTML5.php media_entity_audio_patch_4/src/Plugin/Field/FieldFormatter/AudioPlayerHTML5.php
--- media_entity_audio_patch_3/src/Plugin/Field/FieldFormatter/AudioPlayerHTML5.php	2017-11-06 14:15:55.703244422 +1100
+++ media_entity_audio_patch_4/src/Plugin/Field/FieldFormatter/AudioPlayerHTML5.php	2017-11-06 14:18:04.659263612 +1100
@@ -81,7 +81,7 @@
     foreach ($this->getEntitiesToView($items, $langcode) as $delta => $file) {
       $item = $file->_referringItem;
       $elements[$delta] = array(
-        '#theme' => 'media_file_formatter',
+        '#theme' => 'media_audio_file_formatter',
         '#file' => $file,
         '#description' => $item->description,
         '#value' => $provide_download_link,
diff -ruN media_entity_audio_patch_3/templates/media-audio-file-formatter.html.twig media_entity_audio_patch_4/templates/media-audio-file-formatter.html.twig
--- media_entity_audio_patch_3/templates/media-audio-file-formatter.html.twig	1970-01-01 10:00:00.000000000 +1000
+++ media_entity_audio_patch_4/templates/media-audio-file-formatter.html.twig	2017-11-06 14:18:04.659263612 +1100
@@ -0,0 +1,20 @@
+{#
+/**
+ * @file
+ * Default theme implementation for a link to a file.
+ *
+ * Available variables:
+ * - attributes: The HTML attributes for the containing element.
+ * - link: A link to the file.
+ *
+ * @see template_preprocess_file_link()
+ *
+ * @ingroup themeable
+ */
+#}
+<audio {{ extravalue }}>
+<source src="{{ media_link }}" type="{{ mimetype }}">
+</audio>
+{% if value %}
+<a href="{{ media_link }}" target="_blank" download>{{ 'Download'|t }}</a>
+{% endif %}
diff -ruN media_entity_audio_patch_3/templates/media-file-formatter.html.twig media_entity_audio_patch_4/templates/media-file-formatter.html.twig
--- media_entity_audio_patch_3/templates/media-file-formatter.html.twig	2016-07-31 02:40:39.000000000 +1000
+++ media_entity_audio_patch_4/templates/media-file-formatter.html.twig	1970-01-01 10:00:00.000000000 +1000
@@ -1,21 +0,0 @@
-{#
-/**
- * @file
- * Default theme implementation for a link to a file.
- *
- * Available variables:
- * - attributes: The HTML attributes for the containing element.
- * - link: A link to the file.
- *
- * @see template_preprocess_file_link()
- *
- * @ingroup themeable
- */
-#}
-<audio {{ extravalue }}>
-<source src="{{ media_link }}" type="{{ mimetype }}">
-</audio>
-{% if value %}
-<a href="{{ media_link }}" target="_blank" download>Download</a>
-{% endif %}
-
diff -ruN media_entity_audio_patch_3/templates/media-file-formatter.html.twig.rej media_entity_audio_patch_4/templates/media-file-formatter.html.twig.rej
--- media_entity_audio_patch_3/templates/media-file-formatter.html.twig.rej	1970-01-01 10:00:00.000000000 +1000
+++ media_entity_audio_patch_4/templates/media-file-formatter.html.twig.rej	2017-11-06 14:18:08.815801943 +1100
@@ -0,0 +1,23 @@
+--- templates/media-file-formatter.html.twig
++++ templates/media-file-formatter.html.twig
+@@ -1,20 +0,0 @@
+-{#
+-/**
+- * @file
+- * Default theme implementation for a link to a file.
+- *
+- * Available variables:
+- * - attributes: The HTML attributes for the containing element.
+- * - link: A link to the file.
+- *
+- * @see template_preprocess_file_link()
+- *
+- * @ingroup themeable
+- */
+-#}
+-<audio {{ extravalue }}>
+-<source src="{{ media_link }}" type="{{ mimetype }}">
+-</audio>
+-{% if value %}
+-<a href="{{ media_link }}" target="_blank" download>{{ 'Download'|t }}</a>
+-{% endif %}
